### PR TITLE
Dint 842 removeFromRequisitoinList and requisitionListItems

### PR DESCRIPTION
**Pull request by benjaminkalk**
## Description

[](https://jira.corp.adobe.com/browse/DINT-842)

## Related Issue

https://jira.corp.adobe.com/browse/DINT-842

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
-   [ x My code follows the code style of this project.
-   [x] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [x] I have read the **CONTRIBUTING** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

**Comment by aniham** [comment]
@benjaminkalk this is going to be a pain to diff but can you do the following:


```
cd packages/storefront-events-collector
npm run format:fix
npm run lint:fix
```

same for storefront-event-sdk

**Comment by benjaminkalk** [comment]
@aniham we will do lint stuff as a separate PR

File: packages/storefront-events-collector/src/handlers/requisitionList/removeFromRequisitionListAEP.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    } else {
+        payload = {
+            commerce: {
+                requisitionList: {
+                    ID: requisitionListContext?.id,
+                    name: requisitionListContext?.name,
+                    description: requisitionListContext?.description
+                },
+            },
+            productListItems: requisitionListItemsContext?.items?.map((item) => {
```
Comment (grahamcrackers) [inline packages/storefront-events-collector/src/handlers/requisitionList/removeFromRequisitionListAEP.ts:25]: Should we map this to a function of it's own? This logic is going to be reused quite a bit. We can refactor in another PR.

**Review by grahamcrackers — COMMENTED** [review]
Reviewed. Just a single comment. Everything else seems to be boilerplate.

**Review by bbatsche — APPROVED** [review]

**Review by bbatsche — APPROVED** [review]

File: packages/storefront-events-collector/src/handlers/requisitionList/removeFromRequisitionListAEP.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    } else {
+        payload = {
+            commerce: {
+                requisitionList: {
+                    ID: requisitionListContext?.id,
+                    name: requisitionListContext?.name,
+                    description: requisitionListContext?.description
+                },
+            },
+            productListItems: createProductListItemsFromRequisitionListItems(requisitionListItemsContext, storefrontInstanceContext),
```
Comment (grahamcrackers) [inline packages/storefront-events-collector/src/handlers/requisitionList/removeFromRequisitionListAEP.ts:26]: i'm going to approve this for now, but we should step back and see how to "generic-ize" this as we have multiple places creating `productListItems` from a product array

**Review by grahamcrackers — APPROVED** [review]

**Pull request merged**